nifi-nar-bundles/nifi-standard-services/nifi-lookup-services-bundle/nifi-lookup-services/src/main/java/org/apache/nifi/lookup/db/AbstractDatabaseLookupService.java

```diff
@@ -0,0 +1,100 @@
+            .identifiesControllerService(DBCPService.class)
+            .build();
+
+    static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("dbrecord-lookup-table-name")
+            .displayName("Table Name")
+            .description("The name of the database table to be queried. Note that this may be case-sensitive depending on the database.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .build();
+
+    static final PropertyDescriptor LOOKUP_KEY_COLUMN = new PropertyDescriptor.Builder()
+            .name("dbrecord-lookup-key column")
+            .displayName("Lookup Key Column")
+            .description("The column in the table that will serve as the lookup key. This is the column that will be matched against "
+                    + "the property specified in the lookup processor. Note that this may be case-sensitive depending on the database.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    static final PropertyDescriptor CACHE_SIZE = new PropertyDescriptor.Builder()
+            .name("dbrecord-lookup-cache-size")
+            .displayName("Cache Size")
+            .description("Specifies how many lookup values/records should be cached. The cache is shared for all tables and keeps a map of lookup values to records. "
+                    + "Setting this property to zero means no caching will be done and the table will be queried for each lookup value in each record. If the lookup "
+                    + "table changes often or the most recent data must be retrieved, do not use the cache.")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .defaultValue("0")
+            .required(true)
+            .build();
+
+    static final PropertyDescriptor CLEAR_CACHE_ON_ENABLED = new PropertyDescriptor.Builder()
+            .name("dbrecord-lookup-clear-cache-on-enabled")
+            .displayName("Clear Cache on Enabled")
+            .description("Whether to clear the cache when this service is enabled. If the Cache Size is zero then this property is ignored. Clearing the cache when the "
+                    + "service is enabled ensures that the service will first go to the database to get the most recent data.")
+            .allowableValues("true", "false")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .defaultValue("true")
+            .required(true)
+            .build();
+
+    protected List<PropertyDescriptor> properties;
+
+    protected DBCPService dbcpService;
+
+    protected volatile String lookupKeyColumn;
+
+    protected volatile JdbcCommon.AvroConversionOptions options;
```

Comment:
This AvroConversionOptions is only used when DatabaseRecordLookupService creates schema for resultSetRecordSet. Should be moved from AbstractDatabaseLookupService to DatabaseRecordLookupService.